### PR TITLE
Add step to add PR comments on apply

### PR DESCRIPTION
**Pull request by MoonMoon1919**
# What it is

Adds a step to add a comment to link to apply logs on the merged PR after apply job completes (regardless of success or failure).

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -256,7 +308,9 @@ runs:
         JOB_URL: ${{ steps.jobs_data.outputs.JOB_URL }}
         WORKING_DIR: ${{ inputs.working_directory }}
         ACTOR: ${{ inputs.actor }}
+        PR_NUMBER: ${{ steps.get_pr_number.outputs.get_pr_number }}
       shell: bash
       run: |
-        ISSUE_BODY="Terragrunt Apply failed :x: for "$'`'"$WORKING_DIR"$'`.\n\n Inspect the attached logs, triage the issue, then create a new pull request with the required changes.\n\n[View logs]('"$JOB_URL"$')'
+        ISSUE_BODY="Terragrunt Apply failed :x: for "$'`'"$WORKING_DIR"$'` when applying changes requested in PR #$PR_NUMBER.\n\n _Inspect the attached logs, triage the issue, then create a new pull request with the required changes._\n\n[View logs]('"$JOB_URL"$')'
```
Comment (josh-padnick) [inline action.yml:314]: Is it weird to reference the PR explicitly when the apply comments are in that same PR? Also, we should be consistent with the placement of the emoji.

Perhaps we should say:

:x: Terragrunt Apply failed for `/some/path`. 

_Inspect the logs below, resolve the issue, and then create a new pull request with the required changes._

Comment (MoonMoon1919) [reply action.yml:314]: > Is it weird to reference the PR explicitly when the apply comments are in that same PR?

I was going back and forth. Without the reference to the PR all you get is an issue stating apply failed w/o the context of the origination point of the change. By referencing the PR in the issue GH will also automatically notate the issue in the PR. I thought it was a better ux to make sure they were cross referenced so users would see that we created an issue.

> we should be consistent with the placement of the emoji

Seems reasonable to me, i'll fix it.

Comment (MoonMoon1919) [reply action.yml]: Address in [d4a2b87](https://github.com/gruntwork-io/pipelines-dispatch/pull/10/commits/d4a2b87ad855ef8a9419c19178c2f4cefa316123) and [15e803f](https://github.com/gruntwork-io/pipelines-dispatch/pull/10/commits/15e803fbb509bca240c1d83fa614c23f9b0d42eb)

File: action.yml [hunk 2]
```diff
@@ -226,7 +226,7 @@ runs:
       shell: bash
       run: |
         if [ $TERRAGRUNT_STEP_STATUS = 'succeeded' ]; then
-          COPY="_To **apply** all changes, merge this pull request._"
+          COPY="_To **apply** all changes, merge this pull request._ On merge, another comment will be added to this pull request with apply logs."
```
Comment (josh-padnick) [inline action.yml:229]:
```suggestion
          COPY="_To **apply** all changes, merge this pull request. On merge, another comment will be added to this pull request with apply logs._"
```

File: action.yml (hunk 1 above)
Comment (josh-padnick) [reply action.yml]: Oh, this is a comment in an _issue_, not the original PR. Ok, then yeah, we definitely need the PR link. I know we talked about this, but I couldn't remember the resolution: Is it not possible to add these comments to the PR itself, even if it's already merged and closed?

Comment (MoonMoon1919) [reply action.yml]: You can add the comments to the PR when it's merged close, in this case we're doing both

Comment (josh-padnick) [reply action.yml]: > An argument could be made 

Those words sound like "put it in config, and make it an option to me." I think @ellisonc may have to create a config file concept as part of his work on supporting multiple infra-lives.

Comment (MoonMoon1919) [reply action.yml]: Let's catch up offline about it @josh-padnick

Comment (josh-padnick) [reply action.yml]: Sounds good; we should stay within the scope bounds of this story. But I'm seeing more and more pressure to get a config file in place.

**Review by infraredgirl — APPROVED** [review]
LGTM!

**Review by infraredgirl — APPROVED** [review]

**Review by infraredgirl — APPROVED** [review]

**Pull request merged**